specs/templates/table-column-hld.md

```diff
@@ -0,0 +1,133 @@
+
+- *Relevant historical or background information*
+- *Link to relevant work items, related existing issues, etc.*
+
+### Features
+
+*A list of the key features unique to this component. If adding support for an existing Nimble component's capabilities (e.g. as `nimble-table-column-mapping` does for Nimble icons and `nimble-spinner`), consider which parts of that component's API should be exposed, and whether on a per-column or per-cell basis. Any parts of the API explicitly not supported should be called out below as non-goals.*
+
+### Non-goals
+
+*A list of use cases, features, or functionality which are **not** goals for the component.*
+
+## Implementation / Design
+
+*Describe interesting implementation/design decisions, with justifications and alternates considered.*
+
+*Include code snippets showing basic component use and any interesting configurations.*
+
+Typical usage:
+
+```html
+<nimble-table>
+    <nimble-table-column-*>
+        Column Label
+    </nimble-table-column-*>
+</nimble-table>
+```
+
+### API
+
+#### Column Element
+
+_Element Name_
+
+-   `nimble-table-column-*`
+
+_Props/Attrs_
+
+-   `attr-name`: string (defaults to ...) - Description of the property
+
+_Content_
+
+-   column label (icon and/or text)
+
+_Events_
+
+*List any events along with event detail interface(s).*
+
+#### Cell View Element
+
+_Element Name_
+
+-   `nimble-table-column-*-cell-view`
+
+_Rendering_
+
+*Describe what the cell view will render.*
+
+#### Group Header View Element
+
+*If an existing group header view element will be used instead of creating a new one:*
+
+    A new element will not be created for the group header view. The column will specify the existing `tableColumnTextGroupHeaderViewTag` as the `groupHeaderViewTag` because the header will contain only text.
+
+*Otherwise:*
+
+_Element Name_
+
+-   `nimble-table-column-*-group-header-view`
+
+_Rendering_
+
+*Describe what the group header view will render.*
+
+### Sorting / Grouping
+
+*Will the column be sortable and/or groupable? If so, which `xxx-field-name` attribute will be assigned to  `operandDataRecordFieldName`? If sortable, which `TableColumnSortOperation` will be used? Describe any interesting use cases.*
+
+
+### Sizing
+
+*Will the column use a fractional width, fixed width, or both (via `width-mode` attribute)?*
+
+### Placeholder
+
+*What is the behavior of cells and group rows when the value is `undefined` or `null`? Will a `placeholder` be supported to display a specific interpretation for missing values (e.g. showing "Never run" in a date-time column of latest execution times)? Are there additional values (e.g. empty string) for which placeholders or default rendering need to be considered?*
+
+### Delegated Events
+
+*Will the cells delegate any events to the column element?*
+
+### Focus Recycling
+
+*Will the cell view need to override `focusedRecycleCallback()` to perform any actions when the cell is recycled while it has focus? If so, explain.*
```

Comment:
In a similar vein to the delegated events section, it might be helpful to quickly mention why/when cell recycling happens, and an example of a column that reacts to this. It might be sufficient to simply link to existing docs if they exist.

Reply:
Added a little more detail, and also a link to the issue that might make `focusedRecycleCallback` obsolete.